sycl/test/basic_tests/aspect_usage.cpp

```diff
@@ -0,0 +1,50 @@
+// Test checks Propagate Aspect Usage pass.
```

Comment:
It test warnings emission. I would expect test for warnings in clang tests.

Reply:
Which directory would be the best for such test? I assume `SemaSYCL` is not applicable here, because no changes to Sema were introduced. Perhaps `CodeGenSYCL`?

Reply:
Diagnostic tests are usually added to SemaSYCL

Reply:
> Diagnostic tests are usually added to SemaSYCL

Because those diagnostics are emitted by `Sema`. I'm just not sure that test for a non-`Sema` component should be in `Sema`: we can't even use `-fsyntax-only` in that test. Diagnostics here are essentially emitted from post-`CodeGen`, so to me `CodeGen` seems like the best fit

Reply:
https://reviews.llvm.org/D106030 adds test to clang/test/Frontend directory. I suggest to be consistent with community code and do the same.

Reply:
Added `Frontend` test in 75d53aee043a341cb9448ecdafde583876718efb

Reply:
Perhaps @AaronBallman can answer this as he reviewed https://reviews.llvm.org/D106030. I don't know what is the right directory for the tests checking diagnostics emitted from LLVM passes.

Reply:
Tests in Sema should almost always be `-fsyntax-only` which means we never even emit the LLVM IR let alone run the backend. I think Frontend is a good place for the diagnostic tests as that's what eventually reports them out to the user. I would not make a FrontendSYCL directory as I don't expect us to have a significant number of tests in it. If we find that we do need quite a few tests (a dozen-ish?) then it may make sense to add the directory at that time.